File: hackerrank/easy/0003-save-the-prisoner/js/solution.mjs
// solution.mjs

export function saveThePrisoner(n, m, s) {
  const x = (s + m - 1) % n;
  return x === 0 ? n : x;
}

// --- demo run ---
if (import.meta.url === `file://${process.argv[1]}`) {
  console.log(saveThePrisoner(5, 2, 1)); // 2
  console.log(saveThePrisoner(5, 2, 2)); // 3
  console.log(saveThePrisoner(7, 19, 2)); // 6
  console.log(saveThePrisoner(3, 7, 3)); // 3
}

File: hackerrank/easy/0003-save-the-prisoner/swift/solution.swift
// solution.swift

func saveThePrisoner(n: Int, m: Int, s: Int) -> Int {
    let x = (s + m - 1) % n
    return x == 0 ? n : x
}

// --- demo run ---
print(saveThePrisoner(n: 5, m: 2, s: 1))  // 2
print(saveThePrisoner(n: 5, m: 2, s: 2))  // 3
print(saveThePrisoner(n: 7, m: 19, s: 2)) // 6
print(saveThePrisoner(n: 3, m: 7, s: 3))  // 3